raiden/blockchain_events_handler.py

```diff
@@ -32,44 +31,43 @@
 def handle_tokennetwork_new(raiden, event: Event):
     """ Handles a `TokenNetworkCreated` event. """
     data = event.event_data
-    token_network_address = data['token_network_address']
+    args = data['args']
+    block_number = data['block_number']
+    token_network_address = args['token_network_address']
+    token_address = typing.TokenAddress(args['token_address'])
 
     token_network_proxy = raiden.chain.token_network(token_network_address)
     raiden.blockchain_events.add_token_network_listener(
         token_network_proxy=token_network_proxy,
         contract_manager=raiden.contract_manager,
-        from_block=data['blockNumber'],
-    )
-
-    token_address = typing.TokenAddress(
-        data_decoder(event.event_data['args']['token_address']),
```

Comment:
this is an example of `decode_event_to_internal` failing on it's job, the address has to be decoded there not here.